src/Compilers/Core/Portable/InternalUtilities/ReadOnlyHashSet.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Roslyn.Utilities
+{
+    internal struct ReadOnlyHashSet<T> : IReadOnlySet<T>
```

Comment:
Missed that. Difference is IReadOnlySet has only Count and Contains. I wanted to express that order must not be dependended upon (would break determinism).